pkg/input/remoteread/remoteread.go

```diff
@@ -1,3 +1,242 @@
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/thanos-community/obslytics/pkg/input"
+	"github.com/thanos-community/obslytics/pkg/version"
+	"net/url"
+
+	"time"
+)
+
+type remoteReadInput struct {
+	logger log.Logger
+	conf   input.InputConfig
+}
+
+func NewRemoteReadInput(logger log.Logger, conf input.InputConfig) (remoteReadInput, error) {
+	return remoteReadInput{logger: logger, conf: conf}, nil
+}
+
+// TranslatePromMatchers returns proto matchers (prompb) from Prometheus matchers.
+// NOTE: It allocates memory.
+func TranslatePromMatchers(ms ...*labels.Matcher) ([]*prompb.LabelMatcher, error) {
+	res := make([]*prompb.LabelMatcher, 0, len(ms))
+	for _, m := range ms {
+		var t prompb.LabelMatcher_Type
+
+		switch m.Type {
+		case labels.MatchEqual:
+			t = prompb.LabelMatcher_EQ
+		case labels.MatchNotEqual:
+			t = prompb.LabelMatcher_NEQ
+		case labels.MatchRegexp:
+			t = prompb.LabelMatcher_RE
+		case labels.MatchNotRegexp:
+			t = prompb.LabelMatcher_NRE
+		default:
+			return nil, errors.Errorf("unrecognized matcher type %d", m.Type)
+		}
+		res = append(res, &prompb.LabelMatcher{Type: t, Name: m.Name, Value: m.Value})
+	}
+	return res, nil
+}
+
+func (i remoteReadInput) Open(ctx context.Context, params input.SeriesParams) (input.SeriesIterator, error) {
+
+	tlsConfig := config_util.TLSConfig{
+		CAFile:             i.conf.TLSConfig.CAFile,
+		CertFile:           i.conf.TLSConfig.CertFile,
+		KeyFile:            i.conf.TLSConfig.KeyFile,
+		ServerName:         i.conf.TLSConfig.ServerName,
+		InsecureSkipVerify: i.conf.TLSConfig.InsecureSkipVerify,
+	}
+
+	httpConfig := config_util.HTTPClientConfig{
+		TLSConfig: tlsConfig,
+	}
+
+	parsedUrl, err := url.Parse(i.conf.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+	timeoutDuration, err := model.ParseDuration("10s")
+	if err != nil {
+		return nil, err
+	}
+	endpointUrl := &config_util.URL{URL: parsedUrl}
+	clientConfig := &remote.ClientConfig{
+		URL:              endpointUrl,
+		Timeout:          timeoutDuration,
+		HTTPClientConfig: httpConfig,
+	}
+
+	clientName := "Obslytics/" + version.Version
+	client, err := remote.NewReadClient(clientName, clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	promLabelMatchers, err := TranslatePromMatchers(params.Matchers...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Contruct Query
+	query := &prompb.Query{
+		StartTimestampMs: timestamp.FromTime(params.MinTime),
+		EndTimestampMs:   timestamp.FromTime(params.MaxTime),
+		Matchers:         promLabelMatchers,
+	}
+
+	readResponse, err := client.Read(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+
+	var readSeriesList = make([]ReadSeries, 0, len(readResponse.Timeseries))
```

Comment:
```suggestion
	readSeriesList := make([]ReadSeries, 0, len(readResponse.Timeseries))
```

This is because of consistency, if we allocate we usually use :=